context.c

```diff
@@ -34,6 +34,15 @@ static const char xml_header[] = "<?xml version=\"1.0\" encoding=\"utf-8\"?>"
 "<!ATTLIST buffer-attribute name CDATA #REQUIRED>"
 "]>";
 
+void iio_update_xml_indexes(ssize_t ret, char **ptr, ssize_t *len, ssize_t *alen)
+{
+	if (*ptr) {
+		*ptr += ret;
+		*len -= ret;
+	}
+	*alen += ret;
+}
```

Comment:
I don't like this. The fact that it takes arguments as pointers and update them.

... But I like the +++/--- diff ;)

Maybe you could:
- have it as a static inline in iio-private.h, since this is really just a tiny helper;
- "alen" could be dropped, if the functions keep the original value of ptr as e.g. "buf" and `return (uintptr_t) ptr - (uintptr_t) buf;`. I have a small preference for that, but not a deal-breaker either.

Reply:
I don't like it neither updating variables in functions, but is the best solution that came to my mind to remove that code duplication. 
Regarding your second suggestion, ptr the first time is null so it wont work. I think that's why alen is used.